packages/databricks-vscode/src/configuration/auth/AuthProvider.ts

```diff
@@ -151,33 +149,27 @@ export abstract class AuthProvider {
                 );
 
             case "databricks-cli":
-                if (!config.databricksCliPath) {
-                    throw new Error("Missing path for databricks-cli");
-                }
-
-                return new DatabricksCliAuthProvider(
-                    host,
-                    config.databricksCliPath
-                );
+                return new DatabricksCliAuthProvider(host, cli);
```

Comment:
With so many changes around passing cli to auth providers I'm not sure I can actually understand what exactly we are improving, can you explain it a bit?

Reply:
If the profile does not have a `databricksclipath` variable, we would error out, https://github.com/databricks/databricks-vscode/pull/1114/files#diff-3e8d1317ab3ec8502ceebeae3f641a4e8f7388e442dc8804cdfdb5ea2db283b8L154-L156. 

But, we can be using the internal databricks cli instead of relying on the profile to have the path. It is quiet a common profile pattern to not hard code the cli path. In fact, we should not be writing the cli path either because the profile will fail when we update the extension (hence breaking the path).

Reply:
This was breaking oauth for users with existing profiles.